dotcom-rendering/src/web/components/FixedMediumSlowXIIMPU.tsx

```diff
@@ -0,0 +1,436 @@
+			padSides={true}
+			showDivider={true}
+			showTopMarginWhenStacked={true}
+		>
+			<FrontCard
+				trail={trails[1]}
+				containerPalette={containerPalette}
+				showAge={showAge}
+				trailText={
+					trails[1]?.supportingContent &&
+					trails[1].supportingContent.length > 0
+						? undefined
+						: trails[1].trailText
+				}
+				supportingContent={
+					trails[1].trailText
+						? undefined
+						: trails[1].supportingContent
+				}
+			/>
+		</LI>
+		<LI percentage="33.333%" showDivider={true}>
+			<Hide until="tablet">
+				<AdSlot position="inline" index={index} />
+			</Hide>
+		</LI>
+	</UL>
+);
+
+/**
+ * If the count of cards is odd there is just a single card on
+ * the bottom row with nothing under it which does not need
+ * padding
+ */
+function shouldPadWhenOdd(i: number, noOfCards: number) {
+	return i !== noOfCards - 1;
+}
+
+/**
+ * If the count of cards is even the last two sit on the bottom
+ * row with nothing underneath so we don't want to pad them
+ */
+function shouldPadWhenEven(i: number, noOfCards: number) {
+	return i !== noOfCards - 1 && i !== noOfCards - 2;
+}
+
+/**
+ *
+ * When there is an odd number of cards the last card spans
+ * both columns so we don't want to push the divider for the
+ * previous two cards above it down. If we did it would
+ * touch
+ */
+function shouldOffsetWhenOdd(i: number, noOfCards: number) {
+	return i !== noOfCards - 2 && i !== noOfCards - 3;
+}
+
+/**
+ * FixedMediumSlowXIIMPU
+ *
+ */
+export const FixedMediumSlowXIIMPU = ({
+	trails,
+	containerPalette,
+	showAge,
+	index,
+}: Props) => {
+	switch (trails.length) {
+		case 0: {
+			return null;
+		}
+		case 1: {
+			return (
+				<Card100
+					trails={trails}
+					containerPalette={containerPalette}
+					showAge={showAge}
+				/>
+			);
+		}
+		case 2: {
+			return (
+				<Card50_Card50
+					trails={trails}
+					containerPalette={containerPalette}
+					showAge={showAge}
+				/>
+			);
+		}
+		case 3: {
+			return (
+				<Card33_Card33_Card33
+					trails={trails}
+					containerPalette={containerPalette}
+					showAge={showAge}
+				/>
+			);
+		}
+		case 4: {
+			const topThree = trails.slice(0, 3);
+			const remainingCards = trails.slice(3);
+			return (
+				<>
+					<Card33_Card33_Card33
+						trails={topThree}
+						containerPalette={containerPalette}
+						showAge={showAge}
+					/>
+
+					<Card66_Ad33
+						trails={remainingCards}
+						containerPalette={containerPalette}
+						showAge={showAge}
+						index={index}
+					/>
+				</>
+			);
+		}
+		case 5: {
+			const topThree = trails.slice(0, 3);
+			const remainingCards = trails.slice(3);
+			return (
+				<>
+					<Card33_Card33_Card33
+						trails={topThree}
+						containerPalette={containerPalette}
+						showAge={showAge}
+					/>
+
+					<Card33_Card33_Ad33
+						trails={remainingCards}
+						containerPalette={containerPalette}
+						showAge={showAge}
+						index={index}
+					/>
+				</>
+			);
+		}
+		case 6:
```

Comment:
Are these case statements here meant to show that we explicitly support 6,7,8, and 9 trail containers?

Reply:
Basically, yeah